MaxText/layers/attention_op.py

```diff
@@ -727,7 +730,7 @@ def apply_attention(
               """Decode not supported with flash attention.
                               Use `dot_product` instead."""
           )
-        return self.tpu_flash_attention(query, key, value, decoder_segment_ids, self.attn_logits_soft_cap), None, None
+        return self.tpu_flash_attention(query, key, value, decoder_segment_ids, self.attn_logits_soft_cap, sinks), None, None
       else:
         if model_mode == MODEL_MODE_AUTOREGRESSIVE:
```

Comment:
seams attention sinks is not passed for MODEL_MODE_AUTOREGRESSIVE mode, could you add it too (can be follow up PR).

Reply:
I have added check for GPU with flash attention using sinks.